### DISCUSSION TITLE
Dynamic assign resource on Forward planning with resource constraint

### DISCUSSION THREAD

**Discussion by sgfisme** [frePPLe/frepple#539] [Q&A]
Hi, there,

I config 3 resources with different sizes on the different sites, and have 1 demand that it didn't assign a due date. Can the forward planning assign which resource the demand can use and calculate the fab out date?

**Comment by jdetaeye** [comment]
The due date field can't be left empty, but you can always specify a date in the past.   Frepple will search to satisfy the order with minimal lateness.  This search will include finding the appropriate resource (either based on priority, cost, availability or cost+availability).

**Reply by sgfisme** [reply]
Thanks, @jdetaeye 

I tried to config two daily calendar buckets on the same calendar. One bucket set the value as 1000, and the other set value as 750.

I had one sales order which ordered 1500, and its due date created time for simulating forward scheduling.

The plan result fulfilled 1000 for sales order, and had 500 shortage. Is any possible that 500 shortages shift to the next date for producing?

**Reply by jdetaeye** [reply]
The algorithm will search backward to consume capacity in earlier buckets (until the max-early limit of the resource is reached).  If no capacity is found there the algorithm will search in later buckets.
Yes, this search can divide production across multiple capacity buckets.

I'm not sure why this doesn't happen in your model.  Some hints:   
- From your description it looks like you want a resource of type "buckets" or "buckets_day".   
- If your resource type is "buckets", you need to be very very careful with the definition your calendar buckets. The code needs to see changes in the calendar value to detect your buckets.    (The "buckets_day" resource type automatically generates the correct internal buckets and is not so picky on the calendar format)
